src/parser/strings.rs

```diff
@@ -1,5 +1,18 @@
 use std::collections::HashMap;
 
+#[derive(Clone, Copy, Debug)]
+pub struct StringReference(usize);
+
+impl StringReference {
+    pub fn new(idx: i32) -> Option<StringReference> {
+        if idx.is_negative() {
+            Some(StringReference(idx.abs() as usize))
+        } else {
+            None
+        }
+    }
+}
```

Comment:
This encapsulates the logic around string references. They're always negative and if they're not then they're not a string reference!